src/fileio/union_fstream.cpp

```diff
@@ -82,6 +82,11 @@ union_fstream::union_fstream(std::string url,
       original_input_stream_handle = std::static_pointer_cast<std::istream>(s3stream);
     }
   } else {
+    // Remove the preceeding file:// if it's a local URL.
+    if(boost::algorithm::starts_with(url, "file://")) {
```

Comment:
Protocols are case insensitive according to the [spec](https://tools.ietf.org/html/rfc3986#section-3.1). It looks like the case insensitivity is handled properly in the get_protocol function above - but here, I think we should probably match all case variations of `file://`.

Reply:
Hmmm.  Interesting.  This brings up another point, though, as none of the other code paths (e.g. s3 or http) use type insensitive checking.  I'd be happy to add it here, though, but we should probably make sure that HTTP and others work in a separate issue.

Reply:
👍

Reply:
So, I actually took a stab at fixing this, and we always assume that protocols are lower case throughout the code in fileio/*.  It's quite a bit of work to fix this.  I vote we get this in now, but then file an issue to fix the case insensitive protocols globally.

Reply:
Thanks for investigating. That sounds like the right approach to me.